packages/dotcom-shared/src/mutators.ts

```diff
@@ -56,25 +56,30 @@ export function createMutators(userId: string) {
 			},
 		},
 		file: {
-			insert: async (tx, file: TlaFile) => {
-				assert(file.ownerId === userId, ZErrorCode.forbidden)
-				await assertNotMaxFiles(tx, userId)
-
-				await tx.mutate.file.insert(file)
-			},
 			insertWithFileState: async (
 				tx,
 				{ file, fileState }: { file: TlaFile; fileState: TlaFileState }
 			) => {
 				assert(file.ownerId === userId, ZErrorCode.forbidden)
 				await assertNotMaxFiles(tx, userId)
+				assert(file.id.match(/^[a-zA-Z0-9_-]+$/), ZErrorCode.bad_request)
+				assert(file.id.length <= 32, ZErrorCode.bad_request)
+				assert(file.id.length >= 16, ZErrorCode.bad_request)
+				assert(file.id === fileState.fileId, ZErrorCode.bad_request)
+				assert(fileState.userId === userId, ZErrorCode.forbidden)
```

Comment:
since this pr relies on file ids not having commas, I added these extra checks to ensure it.